### DISCUSSION TITLE
What is the recommended approach to acquire files from share and tools folders?

### DISCUSSION THREAD

**Discussion by meastp** [microsoft/vcpkg#18565] [Q&A]
The recent environment variables for triplets, manifests and soon registries and versions will almost make it possible to use upstream `vcpkg`. There is just one thing missing or not documented (at least I couldn't find it):

What is the recommended way to acquire files from the tools and share folders?

For example, when building `proj4` the library will not work without the `proj.db` and `proj.ini` files.

Currently I have patched `scripts/buildsystems/msbuild/vcpkg.targets` adding:
 ```xml
<ItemGroup>
  <ProjFiles Include="$(VcpkgCurrentInstalledDir)$(VcpkgConfigSubdir)share\proj4\proj.db" Condition="exists('$(VcpkgCurrentInstalledDir)$(VcpkgConfigSubdir)share\proj4\proj.db')" />
</ItemGroup>
<Copy SourceFiles="@(ProjFiles)" DestinationFolder="$(OutDir)" Condition="$(MSBuildProjectName.EndsWith('MyProjectName'))" />
```

But this feels like a hack.

Is there a better way? Perhaps a separate .targets-file in my project as long as `$(VcpkgCurrentInstalledDir)` and `$(VcpkgConfigSubdir)` is available?

What do you recommend, to avoid monkey-patching/forking vcpkg?

**Comment by strega-nil** [comment]
I don't personally know anything about this; @ras0219 could you look at this?

**Comment by psi-mdube** [comment]
I'm interested as well.  We are using the same approach, but now we can't update (from appox. 2021-07-01 to latest vcpkg) as `VcpkgConfigSubdir` is not provided anymore, nor `VcpkgCurrentInstalledDir`.

**Comment by JackBoosY** [comment]
cc @ras0219-msft
